### PR TITLE
Check for gz file existence

### PR DESCRIPTION
**Pull request by dominicschaff**
**Why this change is needed:**

CHIRPS-2.0 Has stopped compressing the tiff files. However the same script should work with or without the gzip compressed files.

**Negative effects of this change:**

### CODE REVIEW

File: deafrica/data/chirps.py [hunk 1]
```diff
@@ -29,6 +30,11 @@
 log.info("Starting CHIRPS downloader")
 
 
+def check_for_compression(href):
```
Comment (alexgleith) [inline deafrica/data/chirps.py]: This doesn't check for compression, it checks the URL is valid

File: deafrica/data/chirps.py [hunk 2]
```diff
@@ -45,6 +51,9 @@ def download_and_cog_chirps(
         # Set up a daily process
         in_file = f"chirps-v2.0.{year}.{month}.{day}.tif.gz"
         in_href = DAILY_URL_TEMPLATE.format(year=year, in_file=in_file)
+        if not check_for_compression(in_href):
+            in_file = f"chirps-v2.0.{year}.{month}.{day}.tif"
+            in_href = DAILY_URL_TEMPLATE.format(year=year, in_file=in_file)
         in_data = f"/vsigzip//vsicurl/{in_href}"
```
Comment (alexgleith) [inline deafrica/data/chirps.py]: If it's not compressed, we need to remove the `/vsigzip/` too.

File: deafrica/data/chirps.py [hunk 3]
```diff
@@ -59,6 +68,9 @@ def download_and_cog_chirps(
         # Set up a monthly process
         in_file = f"chirps-v2.0.{year}.{month}.tif.gz"
         in_href = MONTHLY_URL_TEMPLATE.format(in_file=in_file)
+        if not check_for_compression(in_href):
+            in_file = f"chirps-v2.0.{year}.{month}.tif"
+            in_href = MONTHLY_URL_TEMPLATE.format(in_file=in_file)
         in_data = f"/vsigzip//vsicurl/{in_href}"
```
Comment (alexgleith) [inline deafrica/data/chirps.py]: Again, remove `/vsigzip/`

**Review by alexgleith — CHANGES_REQUESTED** [review]
One minor issue and a couple of bugs :-)

File: deafrica/data/chirps.py (hunk 1 above)
Comment (dominicschaff) [reply deafrica/data/chirps.py]: Updated the function name. I think the new name is clearer

File: deafrica/data/chirps.py (hunk 2 above)
Comment (dominicschaff) [reply deafrica/data/chirps.py]: I have updated this.

File: deafrica/data/chirps.py (hunk 3 above)
Comment (dominicschaff) [reply deafrica/data/chirps.py]: I have updated this.

File: deafrica/data/chirps.py [hunk 4]
```diff
@@ -45,7 +51,10 @@ def download_and_cog_chirps(
         # Set up a daily process
         in_file = f"chirps-v2.0.{year}.{month}.{day}.tif.gz"
         in_href = DAILY_URL_TEMPLATE.format(year=year, in_file=in_file)
-        in_data = f"/vsigzip//vsicurl/{in_href}"
+        if not check_for_url_existence(in_href):
+            in_file = f"chirps-v2.0.{year}.{month}.{day}.tif"
+            in_href = DAILY_URL_TEMPLATE.format(year=year, in_file=in_file)
+        in_data = f"/vsicurl/{in_href}"
```
Comment (alexgleith) [inline deafrica/data/chirps.py]: I don't think you can change the order of these. Can you please add a test, or at least test locally.

Comment (alexgleith) [reply deafrica/data/chirps.py]: Yep, test is failing @dominicschaff. Please ensure your changes have coverage :-)

Comment (dominicschaff) [reply deafrica/data/chirps.py]: Sorry, I wasn't thinking clearly. I will duplicate the tests and have both types

Comment (dominicschaff) [reply deafrica/data/chirps.py]: I fixed this, and added tests that are on a different date. I also added a new docker-compose service that can run the tests. I didn't want to change the current test method however, just to make sure that it still works

File: docker-compose.yml [hunk 5]
```diff
@@ -7,3 +7,10 @@ services:
     volumes:
       - .:/code
     command: ["python --version"]
+  tests:
+    build:
+      context: .
+      target: tests
```
Comment (alexgleith) [inline docker-compose.yml:13]: Love it!

File: deafrica/tests/test_chirps_non_gz.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    DAILY_URL_TEMPLATE,
+    MONTHLY_URL_TEMPLATE,
+    download_and_cog_chirps,
+)
+from deafrica.tests.conftest import TEST_DATA_DIR
+
+from botocore.exceptions import ClientError
+
+
+TEST_BUCKET_NAME = "test-bucket"
```
Comment (alexgleith) [inline deafrica/tests/test_chirps_non_gz.py]: Sorry, but can we include this set of tests in the existing test file, so there's only one file to read through for the tests. This makes for less duplication too, I think.

File: Dockerfile [hunk 7]
```diff
@@ -1,4 +1,4 @@
-FROM osgeo/gdal:ubuntu-small-3.3.3
+FROM osgeo/gdal:ubuntu-small-3.3.3 as base
```
Comment (alexgleith) [inline Dockerfile:1]: Great idea!

**Review by alexgleith — CHANGES_REQUESTED** [review]
Can we merge these new tests in with the existing ones. Just add the new test functions into the exissting file, so it's simpler to find later.

File: deafrica/tests/test_chirps_non_gz.py (hunk 6 above)
Comment (dominicschaff) [reply deafrica/tests/test_chirps_non_gz.py]: I shall merge them

**Review by alexgleith — APPROVED** [review]
Next up is to create a "release" and change the Argo template for this job to use the new image. Build runs fairly quick, but it takes a while to sync to the PDS AWS Account.

**Pull request merged**